### ISSUE TITLE
Package BitShuffle using PyPi

### ISSUE THREAD

**Issue by charlesdaniels** [charlesdaniels/bitshuffle#8]
Official [tutorial](https://wiki.python.org/moin/CheeseShopTutorial#Submitting_Packages_to_the_Package_Index).

**Comment by jyn514** [comment]
Done. You just have to submit using build script (I can't do it because I'm not listed as the author in setup.py).

**Comment by charlesdaniels** [comment]
Really excellent work! I couldn’t have done it better myself. 

I will close this issue out once I perform the upload.

**Comment by charlesdaniels** [comment]
Something that occurs to me is that I think there was something special you had to put in setup.py if you wanted to actually have a “command” linked into $PATH after running pip install. We need to check and make sure that this functionality works before uploading the first release.

**Comment by jyn514** [comment]
I believe this is addressed by `entry_points={'console_scripts': ['bitshuffle=bitshuffle:main']}`

**Comment by jyn514** [comment]
See also https://testpypi.python.org/pypi, which allows setup.py to be tested before submitting. You can also run `python setup.py install` to verify for yourself.

**Comment by jyn514** [comment]
We may want to update our PyPI package at the same time we [create executables](https://github.com/charlesdaniels/bitshuffle/issues/11). Docs:
https://docs.travis-ci.com/user/deployment/pypi/

**Comment by charlesdaniels** [comment]
> We may want to update our PyPI package at the same time we create executables. Docs:
> https://docs.travis-ci.com/user/deployment/pypi/

That would be excellent! In a prefect world, tagging a commit as a release would automagically generate binaries, upload them to the GitHub release page, and simultaneously upload the new version into PyPi. This is the dream of CI/DevOps folks everywhere, since it’s pretty much what *Continuous Delivery* by Jez Humble and David Farley (the bible of CD best practices) refers to as the “deploy button”.

**Comment by jyn514** [comment]
Waiting on https://github.com/charlesdaniels/bitshuffle/issues/11

**Comment by charlesdaniels** [comment]
I have created a PyPi account, I just need to know where / how to package up the credentials so we can upload releases. I think @jyn514  posted instructions for this in another issue, but I cannot find them anymore.

**Comment by jyn514** [comment]
This is mostly done. Instructions for editing `travis.yml` [here](https://docs.travis-ci.com/user/deployment/pypi/). We only need your PyPI credentials.

**Comment by jyn514** [comment]
Bad news: The name `bitshuffle` has [already been taken](https://pypi.python.org/pypi/bitshuffle/0.3.4). According to pypi.org, [names can't be duplicated](https://pypi.org/help/#project-name), and since [the project is maintained](https://github.com/kiyo-masui/bitshuffle), we [will not be given the name](https://www.python.org/dev/peps/pep-0541/#name-conflict-resolution-for-active-projects).

Thoughts? We may have to switch to a different project name.

**Comment by charlesdaniels** [comment]
Huh, I googled for "bitshuffle" before naming it that and did not find
anything. Suggestions for alternate names?

**Comment by jyn514** [comment]
From the top of my head:
- bitascii (or asciibit)
- textshuffle
- transmit
- encode (a little to similar to `uuidencode` IMO)

Thoughts:
- should be short and memorable
- should describe the function
- should not be similar to existing packages
(i.e. `packets` is almost identical to [packet](https://pypi.python.org/pypi/packet/0.4))

Personally, I'm a fan of `transmit`

**Comment by charlesdaniels** [comment]
A good project name must not overlap with any term sufficently common
to appear readily on the first Google Search results page for that
term.

As a specific example, "awesome"[1] is a great software project, but is
difficult to find at all, let alone search for help or debugging
information. 

As a matter of course, any term which is in the English dictionary is
not a good idea.

What about 'sendascii'? This does not appear to be the name of an
existing project. 

[1] - https://awesomewm.org/

**Comment by jyn514** [comment]
It's not bad, I just don't think it's particularly memorable. Mind putting this off until release? Going to ask nerdy friends and family :P

**Comment by charlesdaniels** [comment]
Aye I suppose it is not. No hurry, we don’t need to care until it’s time to upload to pypi. 

- - - - 

This email was sent from my smartphone.
